src/test/scala/chiffreTests/passes/FaultInstrumentationSpec.scala

```diff
@@ -19,12 +19,21 @@ import chiffre.inject.{Injector, IdentityInjector, NoInjectorInfo}
 
 import chisel3._
 import firrtl._
-import firrtl.ir.{Circuit, NoInfo, UnknownType}
+import firrtl.ir._
 import firrtl.analyses.InstanceGraph
 import chisel3.iotesters.ChiselFlatSpec
 import firrtl.annotations.{ComponentName, ModuleName, CircuitName}
 
+import scala.collection.mutable.ArrayBuffer
+
 class FaultInstrumentationSpec extends ChiselFlatSpec {
+  private def collect(connections: ArrayBuffer[Connect])(s: Statement): Statement = {
+    s match {
+      case b: Block => b.mapStmt(collect(connections))
+      case c: Connect => connections += c; c
+      case _ => s
+    }
+  }
```

Comment:
You can use the mappers here to simplify the traversal. How about:

```scala
import firrtl.Mappers._
// ...
  private def collect(connections: ArrayBuffer[Connect])(s: Statement): Statement =
    s map collect(connections) match {
      case c: Connect => connections += c; c
      case sx => sx
    }
```

Reply:
Is `firrtl.Mappers` what allows you to use `Statement map f: Statement => Statement` rather than `Statement.mapStmt(f: Statement => Statement)`?  I'd seen the former in code and haven't figured out how it's possible.

Reply:
Yes, with that in scope, you can FIRRTL IR types get access to custom `map` functions that commonly appear in traversals. Based on the function you give it, it'll do the correct map, so long as it can find the appropriate implicit function to handle that map. Summarily: black magic...